### ISSUE TITLE
! VM is unable to access k8s.gcr.io, you may need to configure a proxy or set --image-repository

### ISSUE THREAD

**Issue by rohit94j** [kubernetes/minikube#7231]
Hi there,

i am getting below error while executing minikube.exe start . Below is the complete output.

C:\Kubernetes>minikube.exe start
* minikube v1.8.2 on Microsoft Windows 10 Home Single Language 10.0.18362 Build 18362
* Automatically selected the virtualbox driver
* Creating virtualbox VM (CPUs=2, Memory=2200MB, Disk=20000MB) ...
! VM is unable to access k8s.gcr.io, you may need to configure a proxy or set --image-repository
* Preparing Kubernetes v1.17.3 on Docker 19.03.6 ...
* Launching Kubernetes ...
* Enabling addons: default-storageclass, storage-provisioner
* Waiting for cluster to come online ...
* Done! kubectl is now configured to use "minikube"

**Comment by DasyDong** [comment]
same issue occurs on my mac 

minikube version: v1.9.2  

```
This VM is having trouble accessing https://k8s.gcr.io
💡  To pull new external images, you may need to configure a proxy: https://minikube.sigs.k8s.io/docs/reference/networking/proxy/
```

**Comment by DasyDong** [comment]
issue occurs when using below commands:
```
 minikube start --docker-env HTTP_PROXY=xxx
```
When I run with splitting two steps , no error occurs 
``` 
export HTTP_PROXY=xxx 
minikube start
```

**Comment by sharifelgamal** [comment]
@DasyDong yeah `--docker-env` is not going to do what you want it to, since you need `HTTP_PROXY` to be a local environment variable, not one inside the container runtime.

Our documentation (https://minikube.sigs.k8s.io/docs/handbook/vpn_and_proxy/) explicitly says exporting the environment variable is the correct idea. We'll fix the broken link in our output.

**Comment by sharifelgamal** [comment]
@rohit94j If you're behind a VPN or proxy, try looking at the link I linked above, otherwise try adding `--image-repository=auto` to your `minikube start` call.

**Comment by DasyDong** [comment]
> @DasyDong yeah `--docker-env` is not going to do what you want it to, since you need `HTTP_PROXY` to be a local environment variable, not one inside the container runtime.
> 
> Our documentation (https://minikube.sigs.k8s.io/docs/handbook/vpn_and_proxy/) explicitly says exporting the environment variable is the correct idea. We'll fix the broken link in our output.

sounds good for me

**Comment by medyagh** [comment]
@rohit94j c@DasyDong do you still need help ? 

Meanwhile have you tried out newest driver Docker Driver with latest version of minikube?
you could try
minikube delete
minikube start --driver=docker

for more information on the docker driver checkout:
https://minikube.sigs.k8s.io/docs/drivers/docker/

**Comment by sharifelgamal** [comment]
I'm going to go ahead and close this, feel free to reopen this if it remains an issue.

**Comment by nandrack** [comment]
Just starting learning k8s.
I still have this issue:
😄  minikube v1.15.1 on Fedora 33
✨  Automatically selected the docker driver
✅  Using image repository 
👍  Starting control plane node minikube in cluster minikube
🔥  Creating docker container (CPUs=2, Memory=2200MB) ...
❗  This container is having trouble accessing https://k8s.gcr.io
💡  To pull new external images, you may need to configure a proxy: https://minikube.sigs.k8s.io/docs/reference/networking/proxy/
🐳  Preparing Kubernetes v1.19.4 on Docker 19.03.13 ...
🔎  Verifying Kubernetes components...
🌟  Enabled addons: storage-provisioner, default-storageclass
🏄  Done! kubectl is now configured to use "minikube" cluster and "default" namespace by default

**Comment by MarkDickinson** [comment]
Same issue, also f33 but using kvm rather than docker. And with more recent versions than above
[mark@vmhost3 bin]$ minikube start
😄  minikube v1.16.0 on Fedora 33
✨  Automatically selected the kvm2 driver
👍  Starting control plane node minikube in cluster minikube
🔥  Creating kvm2 VM (CPUs=2, Memory=4096MB, Disk=20000MB) ...
❗  This VM is having trouble accessing https://k8s.gcr.io
💡  To pull new external images, you may need to configure a proxy: https://minikube.sigs.k8s.io/docs/reference/networking/proxy/
🐳  Preparing Kubernetes v1.20.0 on Docker 20.10.0 ...
    ▪ Generating certificates and keys ...
    ▪ Booting up control plane ...
    ▪ Configuring RBAC rules ...
🔎  Verifying Kubernetes components...
🌟  Enabled addons: storage-provisioner, default-storageclass
💡  kubectl not found. If you need it, try: 'minikube kubectl -- get pods -A'
🏄  Done! kubectl is now configured to use "minikube" cluster and "default" namespace by default

Using a web browser to go to https://k8s.gcr.io simply redirects to https://cloud.google.com/container-registry/
Basically the issue probably should be re-opened as if k8s.gcr.io exists as nothing but a redirect that minkube seems not to handle then minikube should not be coded to use it as a default, whatever the new repository is should be coded as the default.
And is the redirect even going to the correct place ?, while there is the "--image-repository" parameter added for issues like this, using that to specify cloud.google.com/container-registry/ just pulls down the web page data as part of the startup log rather than whatever it is looking for.

**Comment by MarkDickinson** [comment]
Same issue, also f33 but using kvm rather than docker. And with more recent versions than above
[mark@vmhost3 bin]$ minikube start
😄  minikube v1.16.0 on Fedora 33
✨  Automatically selected the kvm2 driver
👍  Starting control plane node minikube in cluster minikube
🔥  Creating kvm2 VM (CPUs=2, Memory=4096MB, Disk=20000MB) ...
❗  This VM is having trouble accessing https://k8s.gcr.io
💡  To pull new external images, you may need to configure a proxy: https://minikube.sigs.k8s.io/docs/reference/networking/proxy/
🐳  Preparing Kubernetes v1.20.0 on Docker 20.10.0 ...
    ▪ Generating certificates and keys ...
    ▪ Booting up control plane ...
    ▪ Configuring RBAC rules ...
🔎  Verifying Kubernetes components...
🌟  Enabled addons: storage-provisioner, default-storageclass
💡  kubectl not found. If you need it, try: 'minikube kubectl -- get pods -A'
🏄  Done! kubectl is now configured to use "minikube" cluster and "default" namespace by default

Using a web browser to go to https://k8s.gcr.io simply redirects to https://cloud.google.com/container-registry/
Basically the issue probably should be re-opened as if k8s.gcr.io exists as nothing but a redirect that minkube seems not to handle then minikube should not be coded to use it as a default, whatever the new repository is should be coded as the default.
And is the redirect even going to the correct place ?, while there is the "--image-repository" parameter added for issues like this, using that to specify cloud.google.com/container-registry/ just pulls down the web page data as part of the startup log rather than whatever it is looking for.

Apologies if this does not replace previous comment; came back to post an update and all old text was still in the comment box.
The latest x86 rpm from https://minikube.sigs.k8s.io/docs/start/ does start correctly on a freshly build VM based on f33.

So the issue is that there are just too many places to download minikube from
It fails when obtained from https://github.com/kubernetes/minikube/releases/tag/v1.16.0
It works when the rpm for the same version from the https://minikube.sigs.k8s.io/docs/start/ page is used.

**Comment by herlon214** [comment]
Exporting `NO_PROXY=localhost,127.0.0.1,10.96.0.0/12,192.168.59.0/24,192.168.39.0/24` fixed this issue for me:

```
This VM is having trouble accessing https://k8s.gcr.io
💡  To pull new external images, you may need to configure a proxy: https://minikube.sigs.k8s.io/docs/reference/networking/proxy/
```

**Comment by Eternal21** [comment]
I've hit this problem while trying to start an existing cluster after my laptop rebooted.  What worked for me was the nuclear option of just deleting the cluster: `minikube delete`, and then creating a new one with: `minikube start`.

**Comment by renkrof** [comment]
I had the same issue.  I tried exporting NO_PROXY as suggested above, but that just led to the following problem.  I ran this command to see what the real problem was an noticed that it was looking for the executable in the wrong location: 

```
$ minikube logs --file=logs.txt
E0817 14:33:25.753188    9209 logs.go:192] command /bin/bash -c "sudo /var/lib/minikube/binaries/v1.24.3/kubectl describe nodes --kubeconfig=/var/lib/minikube/kubeconfig" failed with error: /bin/bash -c "sudo /var/lib/minikube/binaries/v1.24.3/kubectl describe nodes --kubeconfig=/var/lib/minikube/kubeconfig": Process exited with status 1
stdout:

stderr:
error: stat /var/lib/minikube/kubeconfig: no such file or directory
 output: "\n** stderr ** \nerror: stat /var/lib/minikube/kubeconfig: no such file or directory\n\n** /stderr **"

```

[minikube](https://minikube.sigs.k8s.io/docs/commands/start/) documentation shows the option to tell minikube where to look for the binary (eg., --binary-mirror), so using that information, I removed the minikube cluster, and it ran successfully

```
$ minikube delete
🔥  Deleting "minikube" in kvm2 ...
💀  Removed all traces of the "minikube" cluster.

$ minikube start --cpus=4 --memory=6g --addons=ingress --binary-mirror /k8s
😄  minikube v1.26.1 on Redhat 8.5
✨  Automatically selected the kvm2 driver. Other choices: podman, none, ssh
👍  Starting control plane node minikube in cluster minikube
🔥  Creating kvm2 VM (CPUs=4, Memory=6144MB, Disk=20000MB) ...
🌐  Found network options:
    ▪ NO_PROXY=localhost,127.0.0.1,10.96.0.0/12,192.168.59.0/24,192.168.49.0/24,192.168.39.0/24
🐳  Preparing Kubernetes v1.24.3 on Docker 20.10.17 ...
    ▪ env NO_PROXY=localhost,127.0.0.1,10.96.0.0/12,192.168.59.0/24,192.168.49.0/24,192.168.39.0/24
    ▪ Generating certificates and keys ...
    ▪ Booting up control plane ...
    ▪ Configuring RBAC rules ...
🔎  Verifying Kubernetes components...
    ▪ Using image gcr.io/k8s-minikube/storage-provisioner:v5
    ▪ Using image k8s.gcr.io/ingress-nginx/controller:v1.2.1
    ▪ Using image k8s.gcr.io/ingress-nginx/kube-webhook-certgen:v1.1.1
    ▪ Using image k8s.gcr.io/ingress-nginx/kube-webhook-certgen:v1.1.1
🔎  Verifying ingress addon...
🌟  Enabled addons: storage-provisioner, default-storageclass, ingress
🏄  Done! kubectl is now configured to use "minikube" cluster and "default" namespace by default
```

**Comment by Abhishek-569** [comment]
I am getting same error when I tried to run minikube in Ubuntu VM, instead of _This VM has having  trouble_ it says _This container_. Any Solutions?
```
minikube start --kubernetes-version=v1.23.8 --binary-mirror /k8s
😄  minikube v1.29.0 on Ubuntu 22.10
✨  Using the docker driver based on user configuration
📌  Using Docker driver with root privileges
👍  Starting control plane node minikube in cluster minikube
🚜  Pulling base image ...
🔥  Creating docker container (CPUs=2, Memory=2200MB) ...
🌐  Found network options:
    ▪ NO_PROXY=localhost,127.0.0.1,10.96.0.0/12,192.168.59.0/24,192.168.49.0/24,192.168.39.0/24
❗  This container is having trouble accessing https://k8s.gcr.io
💡  To pull new external images, you may need to configure a proxy: https://minikube.sigs.k8s.io/docs/reference/networking/proxy/
🐳  Preparing Kubernetes v1.23.8 on Docker 20.10.23 ...
    ▪ env NO_PROXY=localhost,127.0.0.1,10.96.0.0/12,192.168.59.0/24,192.168.49.0/24,192.168.39.0/24
    ▪ Generating certificates and keys ...
    ▪ Booting up control plane ...
    ▪ Configuring RBAC rules ...
    ▪ Using image gcr.io/k8s-minikube/storage-provisioner:v5
🔎  Verifying Kubernetes components...
🌟  Enabled addons: default-storageclass, storage-provisioner
❗  The image 'yandex/clickhouse:latest' was not found; unable to add it to cache.

❗  /snap/bin/kubectl is version 1.26.1, which may have incompatibilities with Kubernetes 1.23.8.
    ▪ Want kubectl v1.23.8? Try 'minikube kubectl -- get pods -A'
🏄  Done! kubectl is now configured to use "minikube" cluster and "default" namespace by default

```

**Comment by satyampsoni** [comment]
Hi I am having the same issue while working with hyper-v. Pls help.
![image](https://user-images.githubusercontent.com/94950988/222944995-89e01b26-ac3d-4012-99d1-d4803149a795.png)

**Comment by miner58** [comment]
I use same problem, but below link was my solution.
https://minikube.sigs.k8s.io/docs/drivers/hyperv/
I command like this to connect minikube to hyperv external switch.

`minikube start --driver=hyperv --hyperv-use-external-switch`
